File: app/src/main/java/com/konstant/tool/httpsender/api/TranslateApi.java
package com.konstant.tool.httpsender.api;

import com.konstant.tool.httpsender.response.TranslateResponse;
import com.konstant.tool.httpsender.sender.Requester;

public interface TranslateApi {

    @Requester.Method(host = "http://api.fanyi.baidu.com/", path = "api/trans/vip/translate/", response = TranslateResponse.class)
    void getTranslate(@Requester.Parameter("q") String originMsg,
                      @Requester.Parameter("from") String originType,
                      @Requester.Parameter("to") String resultType,
                      @Requester.Parameter("appid") String appid,
                      @Requester.Parameter("salt") String salt,
                      @Requester.Parameter("sign") String sign,
                      Requester.Callback<?> callback);

}

File: tool/src/main/java/com/konstant/tool/lite/network/api/TranslateApi.kt
package com.konstant.tool.lite.network.api

import com.konstant.tool.lite.network.response.TranslateResponse
import retrofit2.http.GET
import retrofit2.http.Query
import io.reactivex.Observable

/**
 * 作者：konstant
 * 时间：2019/10/24 10:54
 * 描述：百度翻译
 */

interface TranslateApi {

    companion object {
        const val HOST = "http://api.fanyi.baidu.com/"
    }

    @GET("api/trans/vip/translate/")
    fun getTranslate(@Query("q") originMsg: String,
                     @Query("from") originType: String,
                     @Query("to") resultType: String,
                     @Query("appid") appid: String,
                     @Query("salt") salt: String,
                     @Query("sign") sign: String): Observable<TranslateResponse>

}